dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/dataitem/ProgramItemStageElement.java

```diff
@@ -102,4 +134,25 @@ public Object getSql( ExprContext ctx, CommonExpressionVisitor visitor )
 
         return column;
     }
+
+    private static boolean isNonDefaultStageOffset( int stageOffset )
+    {
+        return stageOffset != Integer.MIN_VALUE;
+    }
+
+    private static boolean isRepeatableStage( ProgramStageService stageService, String programStageId )
+    {
+        ProgramStage programStage = stageService.getProgramStage( programStageId );
+
+        return programStage != null && programStage.getRepeatable();
+    }
+
+    private static String getErrorMessage( String programStageId )
+    {
+        String errorMessage = "StageOffset is allowed only for repeatable stages";
```

Comment:
Can we use an `ErrorCode` exception here for standardized error responses somehow?

Reply:
done